tests/_helpers/mock.zsh

```diff
@@ -0,0 +1,29 @@
+mock() {
+    local target=$1
+    shift
+
+    echo 0 > ${target}_mock_times
+
+    $target() {
+        local target=${funcstack[1]}
+        local mock_times=$(($(cat -- ${target}_mock_times) + 1))
+        local mock_failfile=${target}_mock_${mock_times}_fail
+        echo $mock_times > ${target}_mock_times
+
+        if (${target}_mock_$mock_times $@ &> $mock_failfile); then
```

Comment:
This was intended to execute a mock function in a subshell but removed as the mock function no longer needs to be executed in a subshell. FYI: Use the plural form (parentheses) because you usually have an opening parenthesis and a closing parenthesis.